src/Passes/Source/Profiles/BaseProfile.cpp

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Passes/ExpandStaticAllocation/ExpandStaticAllocation.hpp"
+#include "Passes/QirAllocationAnalysis/QirAllocationAnalysis.hpp"
+#include "Passes/TransformationRule/TransformationRule.hpp"
+#include "Profiles/BaseProfile.hpp"
+#include "Rules/Factory.hpp"
+
+#include "Llvm/Llvm.hpp"
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    llvm::ModulePassManager BaseProfile::createGenerationModulePass(
+        llvm::PassBuilder&                    pass_builder,
+        llvm::PassBuilder::OptimizationLevel& optimisation_level,
+        bool                                  debug)
+    {
+        auto ret = pass_builder.buildPerModuleDefaultPipeline(optimisation_level);
+        // buildPerModuleDefaultPipeline buildModuleOptimizationPipeline
+        auto function_pass_manager = pass_builder.buildFunctionSimplificationPipeline(
+            optimisation_level, llvm::PassBuilder::ThinLTOPhase::None, debug);
+
+        auto inliner_pass =
+            pass_builder.buildInlinerPipeline(optimisation_level, llvm::PassBuilder::ThinLTOPhase::None, debug);
+
+        // TODO(tfr): Maybe this should be done at a module level
+        function_pass_manager.addPass(ExpandStaticAllocationPass());
+
+        RuleSet rule_set;
+
+        // Defining the mapping
+        auto factory = RuleFactory(rule_set);
+
+        factory.useStaticQubitArrayAllocation();
+        factory.useStaticQubitAllocation();
+        factory.useStaticResultAllocation();
+
+        factory.optimiseBranchQuatumOne();
+        //  factory.optimiseBranchQuatumZero();
+
+        factory.disableReferenceCounting();
+        factory.disableAliasCounting();
+        factory.disableStringSupport();
+
+        function_pass_manager.addPass(TransformationRulePass(std::move(rule_set)));
+
+        // Eliminate dead code
+        function_pass_manager.addPass(llvm::DCEPass());
+        function_pass_manager.addPass(llvm::ADCEPass());
+
+        //  function_pass_manager.addPass(llvm::createCalledValuePropagationPass());
+        // function_pass_manager.addPass(createSIFoldOperandsPass());
+
+        // Legacy passes:
```

Comment:
Which of these lines of code apply to using the legacy pass manager instead?

Reply:
This is just a link to all legacy passes which I use to figure out what the pass name is and subsequently derive the pass name for the new pass manager.

Reply:
There is none of it that is for the legacy pass manager. We do not use that one.